packages/amplify-migration-tests/src/__tests__/migration_tests_v10/git-clone-migration-tests.test.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Tests for headless init/pull workflows on git-cloned projects
+ * These tests exercise workflows that hosting executes during backend builds
+ */
+
+ import {
+  addAuthUserPoolOnly,
+  amplifyPushAuth,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getAmplifyInitConfig, getProjectConfig,
+  getSocialProviders,
+  getTeamProviderInfo, gitCleanFdx,
+  gitCommitAll,
+  gitInit,
+  nonInteractiveInitWithForcePushAttach
+} from '@aws-amplify/amplify-e2e-core';
+import { initJSProjectWithProfileV10 } from '../../migration-helpers-v10/init';
+  
+  describe('attach amplify to git-cloned project', () => {
+    const envName = 'test';
+    let projRoot: string;
+    beforeAll(async () => {
+      projRoot = await createNewProjectDir('clone-test');
+      await initJSProjectWithProfileV10(projRoot, { envName, disableAmplifyAppCreation: false });
+      await addAuthUserPoolOnly(projRoot, {});
+      await amplifyPushAuth(projRoot);
+      await gitInit(projRoot);
+      await gitCommitAll(projRoot);
+    });
+  
+    afterAll(async () => {
+      await deleteProject(projRoot);
+      deleteProjectDir(projRoot);
+    });
+  
+    test('headless init and forcePush when triggers are added', async () => {
+      const { projectName } = getProjectConfig(projRoot);
+      await gitCleanFdx(projRoot);
+
+      const socialProviders = getSocialProviders();
+      const categoriesConfig = {
+        auth: {
+          facebookAppIdUserPool: socialProviders.FACEBOOK_APP_ID,
+          facebookAppSecretUserPool: socialProviders.FACEBOOK_APP_SECRET,
+          googleAppIdUserPool: socialProviders.GOOGLE_APP_ID,
+          googleAppSecretUserPool: socialProviders.GOOGLE_APP_SECRET,
+          // eslint-disable-next-line spellcheck/spell-checker
+          loginwithamazonAppIdUserPool: socialProviders.AMAZON_APP_ID,
+          // eslint-disable-next-line spellcheck/spell-checker
+          loginwithamazonAppSecretUserPool: socialProviders.AMAZON_APP_SECRET,
+        },
+      };
+      // checks amplify hosting forcePush on existing projects with v10.5.1
+      expect( () => nonInteractiveInitWithForcePushAttach(projRoot, getAmplifyInitConfig(projectName, envName), categoriesConfig, false)).not.toThrow();
```

Comment:
I'm not sure, but I think this is missing an `await`?

I think it's easier to understand if it's just invoked directly. If it throws, the test will still fail:

```suggestion
      await nonInteractiveInitWithForcePushAttach(projRoot, getAmplifyInitConfig(projectName, envName), categoriesConfig, false);
```